src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -163,19 +163,10 @@ protected static int CompareRange(LSP.Range r1, LSP.Range r2)
             return compareLine != 0 ? compareLine : compareChar;
         }
 
-        protected static string ApplyTextEdits(LSP.TextEdit[] edits, SourceText originalMarkup)
+        protected static string ApplyTextEdits(LSP.TextEdit[]? edits, SourceText originalMarkup)
         {
-            var text = originalMarkup;
-            foreach (var edit in edits)
```

Comment:
📝 This method iterated from start to end, so edits that spanned across the end of a line were corrupted. The new code uses our existing `WithChanges` method that does not restrict change spans in this manner.

Reply:
this is a little inconsistent with the LSP model for text edits.  Those LSP edits are intended to be applied linearly, not in one swell foop (at least for the document/didChange request).  So I'm wary about making this change.

Are we sure these are teh right semantics here?

Reply:
sounds good.  thanks for checking :)  i wish lsp were consistent here :)